modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/ContributorsBuilder.es.js

```diff
@@ -0,0 +1,282 @@
+				};
+			}
+		);
+		this.onCriteriaEdit = this.onCriteriaEdit.bind(this);
+		this.onCriteriaChange = this.onCriteriaChange.bind(this);
+		this.state = {
+			conjunctionName: 'and',
+			contributors,
+			editing: undefined,
+			newPropertyKey: this.props.propertyGroups.length &&
+				this.props.propertyGroups[0].propertyKey
+		};
+		this._handleRootConjunctionClick = this._handleRootConjunctionClick.bind(this);
+		this._handleSelectorChange = this._handleSelectorChange.bind(this);
+	}
+
+	onCriteriaEdit(id, editing) {
+		this.setState(
+			{
+				editing: editing ? undefined : id
+			}
+		);
+	}
+
+	_handleSelectorChange(e) {
+		const newPropertyKey = e.target.value;
+
+		this.setState(
+			prevState => (
+				{
+					...prevState,
+					newPropertyKey
+				}
+			)
+		);
+	}
+
+	onCriteriaChange(criteriaChange, index) {
+		this.setState(
+			state => {
+				let newState;
+				if (state.editing !== index) {
+					newState = state;
+				}
+				else {
+					newState = {
+						contributors: state.contributors.map(
+							(c, i) => {
+								return (index === i) ? {
+									...c,
+									criteriaMap: criteriaChange,
+									query: buildQueryString([criteriaChange])
+								} : c;
+							}
+						)
+					};
+				}
+				return newState;
+			}
+		);
+	}
+
+	_handleRootConjunctionClick(event) {
+		event.preventDefault();
+
+		this.setState(
+			(prevState, prevProps) => {
+				const prevContributors = prevState.contributors;
+
+				const prevConjunction = prevContributors[0] && prevContributors[0].conjunctionId;
+
+				const {supportedConjunctions} = prevProps;
+				const conjunctionIndex = supportedConjunctions.findIndex(
+					item => item.name === prevConjunction
+				);
+				const conjunctionSelected = (conjunctionIndex === supportedConjunctions.length - 1) ?
+					supportedConjunctions[0].name :
+					supportedConjunctions[conjunctionIndex + 1].name;
+
+				const contributors = prevContributors.map(
+					c => ({
+						...c,
+						conjunctionId: conjunctionSelected
+					})
+				);
+
+				return {
+					...prevState,
+					contributors
+				};
+			}
+		);
+	}
+
+	render() {
+		const {
+			propertyGroups,
+			supportedConjunctions,
+			supportedOperators,
+			supportedPropertyTypes
+		} = this.props;
+		const currentEditing = this.state.editing;
+		const selectedContributor = this.state.contributors[currentEditing];
+		const selectedProperty = selectedContributor && propertyGroups.find(c => selectedContributor.propertyKey === c.propertyKey);
+
+		return (
+			<div className={this.classes}>
+				<div className="criteria-builder-section-main">
+					{
+						this.state.contributors.map(
+							(criteria, i) => {
+								return (
+									<React.Fragment key={i}>
+										<div className={'sheet-lg'}>
+											{
+												(i !== 0) &&
+												<React.Fragment>
+													<Conjunction
+														className={'ml-0'}
+														conjunctionName={criteria.conjunctionId}
+														editing={true}
```

Comment:
You shouldn't have to pass `{true}` here. If the prop is present, it will evaluate to true.
`<SomeComponent editing /> === <SomeComponent editing={true} />`